scipy/stats/_distn_infrastructure.py

```diff
@@ -1252,12 +1252,17 @@ def entropy(self, *args, **kwds):
         place(output, cond0, self.vecentropy(*goodargs) + log(goodscale))
         return output
 
-    def moment(self, n, *args, **kwds):
-        """n-th order non-central moment of distribution.
+    def moment(self, order=None, *args, **kwds):
+        """non-central moment of distribution of specified order.
+
+        .. deprecated:: 1.8.0
```

Comment:
Looks like this will need to change to 1.9.0 everywhere, and 1.10.0 will need to be 1.11.0

Reply:
In recent deprecations we put 2.0. I am not sure if we decided yet if we are going to have 1.10 or go directly to 2.

Reply:
I'm not sure what the plan is there. We could say "two releases" instead of assigning it a number.

Reply:
How about "deprecated in 1.9.0 and will be removed two releases after"

Reply:
Fine with me 👍

Reply:
Thanks @ev-br for your revie! You suggested we merge as is - if we do that, should we ask for a backport?

Reply:
Not sure about backporting it. This PR fixes a long-standing problem, sure. But I've hard time justifying it to be release-critical. Also it's complicated enough to probably warrant an rc, and 1.8.x is delayed enough already. All in all, my suggestion is to merge it for 1.9.x early in the release cycle.

Reply:
```suggestion
        .. deprecated:: 1.9.0
```

Reply:
Or today. Should be all set now after 547d1bb522562a1ba38961d13932fffc2bb92edf.

Reply:
@ev-br Wanted to make I understood - you suggested we "merge it for 1.9.x early in the release cycle." Does that mean we are waiting for something?

Reply:
Quite the reverse, the suggestion is to merge it once the deprecation wording is tweaked. And since it is now, I pressed the green button.

Reply:
Thanks @ev-br!